src/Controls/tests/SourceGen.UnitTests/SourceGenXamlTests.cs

```diff
@@ -199,12 +252,69 @@ public void TestCodeBehindGenerator_ModifiedXaml()
 		{
 			{ TrackingNames.ProjectItemProvider, IncrementalStepRunReason.Modified },
 			{ TrackingNames.XamlProjectItemProvider, IncrementalStepRunReason.Modified },
-			{ TrackingNames.ReferenceCompilationProvider, IncrementalStepRunReason.Cached },
+			{ TrackingNames.ReferenceCompilationProvider, IncrementalStepRunReason.Unchanged },
 			{ TrackingNames.XmlnsDefinitionsProvider, IncrementalStepRunReason.Cached },
 			{ TrackingNames.ReferenceTypeCacheProvider, IncrementalStepRunReason.Cached },
 			{ TrackingNames.XamlSourceProvider, IncrementalStepRunReason.Modified }
 		};
 
 		VerifyStepRunReasons(result2, expectedReasons);
 	}
+
+	[Test]
+	public void TestCodeBehindGenerator_NotXaml()
+	{
+		var xaml =
+"""
+<?xml version="1.0" encoding="UTF-8"?>
```

Comment:
The XML declaration is optional or mandatory?

Reply:
when your file encoding is "correct", it can be skipped